src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -109,32 +109,50 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
         {
             case 1:
             {
-                genConsumeOperands(node);
-                op1Reg = op1->gtRegNum;
-
                 if (node->OperIsMemoryLoad())
                 {
-                    emit->emitIns_R_AR(ins, simdSize, targetReg, op1Reg, 0);
-                }
-                else if ((category == HW_Category_SIMDScalar) && HWIntrinsicInfo::CopiesUpperBits(intrinsicId))
-                {
-                    emit->emitIns_SIMD_R_R_R(ins, simdSize, targetReg, op1Reg, op1Reg);
-                }
-                else if ((ival != -1) && varTypeIsFloating(baseType))
-                {
-                    assert((ival >= 0) && (ival <= 127));
-                    genHWIntrinsic_R_RM_I(node, ins, (int8_t)ival);
+                    genConsumeAddress(op1);
```

Comment:
I believe you mentioned that `genConsumeOperands` will also handle the address case, but `genConsumeAddress` is more efficient.

My question then would be: Is it actually worth doing explicit `genConsumeAddress` calls here? It looks like it makes the overall logic slightly more complex...

Reply:
In what way does it make the logic more complex? It's probably a matter of taste, but this case needs special handling anyway, and it seems clearest to explicitly consume an address.

Reply:
>  It's probably a matter of taste

Possibly. I would have thought that the existing `genConsumeOperands(node)` call would have handled things, in which case the changes could have been minimized to just changing `emit->emitIns_R_AR(ins, simdSize, targetReg, op1Reg, 0);` to:
```
// Until we improve the handling of addressing modes in the emitter, we'll create a
// temporary GT_IND to generate code with.
GenTreeIndir load = indirForm(node->TypeGet(), op1);
emit->emitInsLoadInd(ins, simdSize, node->gtRegNum, &load);
```

Likewise, down on L144, just the if block could have been added and the existing `genConsumeOperands(node)` would have handle both the memory store and non-memory store cases (rather than splitting this into 4 separate calls).

Reply:
Please avoid `genConsumeOperands` in SIMD and HWIntrinsic code, it's incompatible with my pending `GT_LIST` changes.